### PR TITLE
feat(infra, ci): Deploy helm charts via ci

### PR DESCRIPTION
**Pull request by aarushik93**
### Background

Deploy helm charts automatically when changes are made in builder or server

### Changes 🏗️

Added helm steps in GHA workflow

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [ ] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

___

### **PR Type**
enhancement, configuration changes

___

### **Description**
- Added new GitHub Actions workflows for building and deploying AutoGPT Builder and Server Docker images using Helm.
- Removed old GitHub Actions workflows for building and pushing AutoGPT Builder and Server Docker images.
- Updated Terraform configuration to include new role bindings for service accounts.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>autogpt-builder-cd.yml</strong><dd><code>Add CI workflow for AutoGPT Builder Docker deployment</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.github/workflows/autogpt-builder-cd.yml

<li>Added a new GitHub Actions workflow for building and deploying AutoGPT <br>Builder Docker images.<br> <li> Configured steps for checking out code, setting up Cloud SDK, <br>configuring Docker, building and pushing Docker images, and deploying <br>to GKE using Helm.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7754/files#diff-9d05c9b2c3454b036cc85674c47a91181f8eea1f5efdcba0b8934182cad9cfd9">+60/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>autogpt-server-cd.yml</strong><dd><code>Add CI workflow for AutoGPT Server Docker deployment</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.github/workflows/autogpt-server-cd.yml

<li>Added a new GitHub Actions workflow for building and deploying AutoGPT <br>Server Docker images.<br> <li> Configured steps for checking out code, setting up Cloud SDK, <br>configuring Docker, building and pushing Docker images, and deploying <br>to GKE using Helm.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7754/files#diff-60dab13eb41b71bd685eae9421d6f88f719644b1c07824f6e13f1b2b55613587">+60/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>autogpt-builder-docker.yml</strong><dd><code>Remove old CI workflow for AutoGPT Builder Docker</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.github/workflows/autogpt-builder-docker.yml

<li>Removed the old GitHub Actions workflow for building and pushing <br>AutoGPT Builder Docker images.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7754/files#diff-24f1e68c85d4c233769ad0faad31f1f7a3af6309107757daace31642779c0aed">+0/-41</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>autogpt-server-docker.yml</strong><dd><code>Remove old CI workflow for AutoGPT Server Docker</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.github/workflows/autogpt-server-docker.yml

<li>Removed the old GitHub Actions workflow for building and pushing <br>AutoGPT Server Docker images.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7754/files#diff-0c7741834ad53d5fd8f82052fad6841f6d10ece990194226c9131a0ee2d7820f">+0/-41</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>dev.tfvars</strong><dd><code>Update Terraform role bindings for service accounts</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/terraform/environments/dev.tfvars

<li>Added new role bindings for service accounts in the Terraform <br>configuration.<br> <li> Included roles for container developer, storage object viewer, and <br>artifact registry writer.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7754/files#diff-afbe1103c303a8001bc8c27865476c0e3d97707e560b5b023a627798504f5e38">+10/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by aarushik93** [comment]
reminder for myself to reapply the terraform and update the secret again as the tf has changed

**Review by ntindle — APPROVED** [review]

File: .github/workflows/autogpt-server-cd.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          project_id: ${{ env.PROJECT_ID }}
+          service_account_key: ${{ secrets.GCP_SA_KEY }}
+          export_default_credentials: true
+
+      - name: Configure Docker
+        run: gcloud auth configure-docker ${{ env.REGION }}-docker.pkg.dev
+
+      - name: Build Docker image
+        working-directory: rnd/autogpt_server
+        run: docker build -t ${{ env.REGION }}-docker.pkg.dev/${{ env.PROJECT_ID }}/${{ env.IMAGE_NAME }}:${{ github.sha }} .
```
Comment (Swiftyos) [inline .github/workflows/autogpt-server-cd.yml:38]: Are we able to add a step to populate the cache before building so we can reuse costly layers - such as linux update layer and poetry install layer - we would also need to update our docker file structure to support this.

**Comment by Swiftyos** [comment]
@aarushik93 What is the status of this PR?

**Comment by aarushik93** [comment]
I need to rebase and update this PR

**Comment by aarushik93** [comment]
Ditto

**Pull request closed without merging**